apps/dashboard/src/components/primitives/field-editor/variable-popover/variable-popover.tsx

```diff
@@ -0,0 +1,263 @@
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+} from '@/components/primitives/command';
+import { FormControl, FormItem } from '@/components/primitives/form/form';
+import { Input, InputField } from '@/components/primitives/input';
+import { Popover, PopoverContent, PopoverTrigger } from '@/components/primitives/popover';
+import { Switch } from '@/components/primitives/switch';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { RiAddFill } from 'react-icons/ri';
+import { useDebounce } from '../../../../hooks/use-debounce';
+import { Code2 } from '../../../icons/code-2';
+import { Separator } from '../../separator';
+import { TransformerItem } from './components/transformer-item';
+import { TransformerList } from './components/transformer-list';
+import { useTransformerManager } from './hooks/use-transformer-manager';
+import { useVariableParser } from './hooks/use-variable-parser';
+import type { VariablePopoverProps } from './types';
+import { formatLiquidVariable } from './utils';
+
+export function VariablePopover({ variable, onUpdate }: VariablePopoverProps) {
+  const { parsedName, parsedDefaultValue, parsedTransformers, originalVariable } = useVariableParser(variable || '');
+  const [name, setName] = useState(parsedName);
+  const [defaultVal, setDefaultVal] = useState(parsedDefaultValue);
+  const [showRawLiquid, setShowRawLiquid] = useState(false);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [isCommandOpen, setIsCommandOpen] = useState(false);
+
+  useEffect(() => {
+    setName(parsedName);
+    setDefaultVal(parsedDefaultValue);
+  }, [parsedName, parsedDefaultValue]);
+
+  const updateVariable = useCallback(
+    (newName: string, newDefaultVal: string, newTransformers: any[]) => {
+      onUpdate(formatLiquidVariable(newName, newDefaultVal, newTransformers));
+    },
+    [onUpdate]
+  );
+
+  const debouncedUpdate = useDebounce(updateVariable, 300);
+
+  const {
+    transformers,
+    dragOverIndex,
+    draggingItem,
+    setDragOverIndex,
+    setDraggingItem,
+    handleTransformerToggle,
+    moveTransformer,
+    handleParamChange,
+    getFilteredTransformers,
+  } = useTransformerManager({
+    initialTransformers: parsedTransformers,
+    onUpdate: (newTransformers) => {
+      debouncedUpdate(name, defaultVal, newTransformers);
+    },
+  });
+
+  const handleNameChange = useCallback(
+    (newName: string) => {
+      setName(newName);
+      debouncedUpdate(newName, defaultVal, transformers);
+    },
+    [defaultVal, transformers, debouncedUpdate]
+  );
+
+  const handleDefaultValueChange = useCallback(
+    (newDefaultVal: string) => {
+      setDefaultVal(newDefaultVal);
+      debouncedUpdate(name, newDefaultVal, transformers);
+    },
+    [name, transformers, debouncedUpdate]
+  );
+
+  const handleRawLiquidChange = useCallback((value: string) => {
```

Comment:
Shouldn't this code be part of the useVariableParser logic? Ideally, we should centralize all the logic about variable parsing and formatting to a single file, so that if we need to modify or expand the syntax, we do it in one place. Currently the logic is scattered across files.